### ISSUE TITLE
Retrofit Crash when analytics are enabled

### ISSUE THREAD

**Issue by ghostwan** [Flagsmith/flagsmith-kotlin-android-client#44]
Library: 1.4.0
Android: 14
Model: Pixel 7

If I do that, it works

```kotlin
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)       
        initFlagsmith()
    }
    private fun initFlagsmith() {
        flagsmith = Flagsmith(environmentKey = "", context = this, enableAnalytics = false)
    }
```

but if I let the analytics enabled 

```kotlin
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)       
        initFlagsmith()
    }
    private fun initFlagsmith() {
        flagsmith = Flagsmith(environmentKey = "", context = this)
    }
```

I have a crash at app startup

```log
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.netatmo.camera/com.netatmo.camera.MainComposeActivity}: kotlin.UninitializedPropertyAccessException: lateinit property retrofit has not been initi
                            alized
                         E      at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3822)
                         E      at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3963)
                         E      at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
                         E      at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:139)
                         E      at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:96)
                         E      at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2468)
                         E      at android.os.Handler.dispatchMessage(Handler.java:106)
                         E      at android.os.Looper.loopOnce(Looper.java:205)
                         E      at android.os.Looper.loop(Looper.java:294)
                         E      at android.app.ActivityThread.main(ActivityThread.java:8248)
                         E      at java.lang.reflect.Method.invoke(Native Method)
                         E      at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
                         E      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
                         E  Caused by: kotlin.UninitializedPropertyAccessException: lateinit property retrofit has not been initialized
                         E      at com.flagsmith.Flagsmith.<init>(Flagsmith.kt:50)
                         E      at com.flagsmith.Flagsmith.<init>(Flagsmith.kt:30)
                         E      at com.netatmo.camera.MainComposeActivity.getFlags(MainComposeActivity.kt:143)
                         E      at com.netatmo.camera.MainComposeActivity.onCreate(MainComposeActivity.kt:139)
                         E      at android.app.Activity.performCreate(Activity.java:8621)
                         E      at android.app.Activity.performCreate(Activity.java:8599)
                         E      at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1456)
                         E      at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3804)
                         E      ... 12 more
```

**Comment by matthewelwell** [comment]
Hi @ghostwan , thanks for this. I believe this is resolved by [this PR](https://github.com/Flagsmith/flagsmith-kotlin-android-client/pull/40) we merged earlier. I'll let you know once it's released.

**Comment by matthewelwell** [comment]
This is resolved in v1.5.0
